remoteappmanager/handlers/base_handler.py

```diff
@@ -40,3 +41,20 @@ def render(self, template_name, **kwargs):
 
         args.update(kwargs)
         super(BaseHandler, self).render(template_name, **args)
+
+    def write_error(self, status_code, **kwargs):
+        """Render error page for uncaught errors"""
+        status_message = responses.get(status_code, 'Unknown HTTP Error')
+        message = ""
+
+        # If this error was caused by an uncaught exception
+        # log exception message and reference number as well
+        exc_info = kwargs.get('exc_info')
+        if exc_info:
+            exception = exc_info[1]
+            ref = self.log.issue(status_message, exception)
+            reason = getattr(exception, 'reason', '')
+            message = '{} Ref.: {}'.format(reason, ref)
```

Comment:
If renaming`reason` to something e.g.`user_message` makes the usage here more reasonable, then it is not the problem of the usage, it is a problem with naming.